### PR TITLE
Fixed copy size and row stride calculations

### PR DESCRIPTION
**Pull request by chaoticbob**
This chagne fixes an issue with the CopyBitmapToImage function that corrects an assumption that the number of pixels copied is always the row stride.

- Fixed row stride calculation for staging buffer.
- Changed number of bytes copied to b based on the bitmap's width and not the row bytes.

### CODE REVIEW

File: src/ppx/graphics_util.cpp [hunk 1]
```diff
@@ -102,17 +102,28 @@ Result CopyBitmapToImage(
...
-    uint32_t alignedRowStride    = RoundUp<uint32_t>(pBitmap->GetRowStride(), rowStrideAlignement);
+    // This is the number of bytes we're going to copy per row.
+    uint32_t rowCopySize = pBitmap->GetWidth() * pBitmap->GetPixelStride();
+
+    // When copying from a buffer to a image/texture, D3D12 requires that the rows
+    // stored in the source buffer (aka staging buffer) are aligned to 256 bytes.
+    // Vulkan does not have this requirement. So for the staging buffer, we want
+    // to enforce the alignment for D3D12 but not for Vulkan.
+    //
+    uint32_t apiRowStrideAlignement = grfx::IsDx12(pQueue->GetDevice()->GetApi()) ? PPX_D3D12_TEXTURE_DATA_PITCH_ALIGNMENT : 1;
```
Comment (vettoreldaniele) [inline src/ppx/graphics_util.cpp:113]: typo `alignment` (also in the comment, line below)

File: src/ppx/graphics_util.cpp [hunk 2]
```diff
@@ -102,17 +102,28 @@ Result CopyBitmapToImage(
...
+    // The staging buffer's row stride alignemnt needs to be based off the bitmap's
+    // width (i.e. the number of bytes we're going to copy) and not the bitmap's row
+    // stride. The bitmap's may be padded beyond width * pixel stride.
+    //
+    uint32_t stagingBufferignedRowStride = RoundUp<uint32_t>(rowCopySize, apiRowStrideAlignement);
```
Comment (vettoreldaniele) [inline src/ppx/graphics_util.cpp:118]: typo `stagingBufferignedRowStride` -> `stagingBufferAlignedRowStride`

**Review by vettoreldaniele — APPROVED** [review]
LGTM with a few nits

File: src/ppx/graphics_util.cpp (hunk 2 above)
Comment (chaoticbob) [reply src/ppx/graphics_util.cpp:118]: Thanks. Changed this to `stagingBufferRowStride`.

File: src/ppx/graphics_util.cpp (hunk 1 above)
Comment (chaoticbob) [reply src/ppx/graphics_util.cpp:113]: Thanks. Fixed.

**Pull request merged**